### ISSUE TITLE
Relative scaling bug

### ISSUE THREAD

**Issue by mikael10j** [vuillaut/ctaplot#46]
If a bin is empty, relative scaling fails due to np.max for s4:

KeyError                                  Traceback (most recent call last)
<ipython-input-22-bc71efa66f78> in <module>
----> 1 bins, res = ctaplot.resolution_per_bin(exp.gamma_data.mc_energy, exp.gamma_data.mc_energy, exp.gamma_data.reco_energy, relative_scaling_method='s1', bins=19)
      2 plt.figure()
      3 plt.errorbar(ctaplot.ana.logbin_mean(bins), res[:,0], xerr=(bins[1:] - bins[:-1])/2., yerr=(res[:,0] - res[:,1], res[:,2] - res[:,0]), fmt='o')

~/anaconda3/envs/torch12/lib/python3.6/site-packages/ctaplot/ana.py in resolution_per_bin(x, y_true, y_reco, percentile, confidence_level, bias_correction, relative_scaling_method, bins)
    366                               confidence_level=confidence_level,
    367                               relative_scaling_method=relative_scaling_method,
--> 368                               bias_correction=bias_correction,
    369                               )
    370                    )

~/anaconda3/envs/torch12/lib/python3.6/site-packages/ctaplot/ana.py in resolution(simu, reco, percentile, confidence_level, bias_correction, relative_scaling_method)
    324         reco_corr = reco - b
    325         res = np.nan_to_num(np.abs((reco_corr - simu) /
--> 326                                    relative_scaling(simu, reco_corr, method=relative_scaling_method)))
    327 
    328     return np.append(_percentile(res, percentile), percentile_confidence_interval(res,

~/anaconda3/envs/torch12/lib/python3.6/site-packages/ctaplot/ana.py in relative_scaling(simu, reco, method)
    285         's2': np.abs(reco),
    286         's3': (np.abs(simu) + np.abs(reco))/2.,
--> 287         's4': np.max([np.abs(reco), np.abs(simu)], axis=0)
    288     }
    289 

<__array_function__ internals> in amax(*args, **kwargs)

~/anaconda3/envs/torch12/lib/python3.6/site-packages/numpy/core/fromnumeric.py in amax(a, axis, out, keepdims, initial, where)
   2619     """
   2620     return _wrapreduction(a, np.maximum, 'max', axis, None, out,
-> 2621                           keepdims=keepdims, initial=initial, where=where)
   2622 
   2623 

~/anaconda3/envs/torch12/lib/python3.6/site-packages/numpy/core/fromnumeric.py in _wrapreduction(obj, ufunc, method, axis, dtype, out, **kwargs)
     88                 return reduction(axis=axis, out=out, **passkwargs)
     89 
---> 90     return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
     91 
     92 

~/anaconda3/envs/torch12/lib/python3.6/site-packages/pandas/core/series.py in __getitem__(self, key)
   1066         key = com.apply_if_callable(key, self)
   1067         try:
-> 1068             result = self.index.get_value(self, key)
   1069 
   1070             if not is_scalar(result):

~/anaconda3/envs/torch12/lib/python3.6/site-packages/pandas/core/indexes/base.py in get_value(self, series, key)
   4728         k = self._convert_scalar_indexer(k, kind="getitem")
   4729         try:
-> 4730             return self._engine.get_value(s, k, tz=getattr(series.dtype, "tz", None))
   4731         except KeyError as e1:
   4732             if len(self) > 0 and (self.holds_integer() or self.is_boolean()):

pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_value()

pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_value()

pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_loc()

pandas/_libs/hashtable_class_helper.pxi in pandas._libs.hashtable.Int64HashTable.get_item()

pandas/_libs/hashtable_class_helper.pxi in pandas._libs.hashtable.Int64HashTable.get_item()

KeyError: 0

**Comment by vuillaut** [comment]
Not sure I could reproduce the specific error you are having.
It might be solved by #47 - could you please check ?

In all case, resolution with empty bins works:
```
size = 10
x = np.linspace(0, 10, size)
bins = np.array([1, 5, 11, 15])
y_true = np.ones(size)
y_reco = np.ones(size)
for scaling in ['s0', 's1', 's2', 's3', 's4']:
    bins, res = ana.resolution_per_bin(x, y_true, y_reco,
                                       bins=bins,
                                       relative_scaling_method=scaling,
                                       bias_correction=True)
```

**Comment by vuillaut** [comment]
The issue arise when using pandas series instead of numpy arrays.
